packages/typespec-azure-resource-manager/lib/Legacy/decorator.tsp

```diff
@@ -7,3 +7,9 @@ namespace Azure.ResourceManager.Legacy;
  * but need to be identified as such.
  */
 extern dec customAzureResource(target: Model);
+
+/**
+ * Specify an external reference that should be used when emitting this type.
+ * @param jsonRef - External reference(e.g. "../../common.json#/definitions/Foo")
+ */
+extern dec commonTypesRef(entity: Model | ModelProperty, jsonRef: valueof string);
```

Comment:
According to the description and our scenarios, I feel like `externalTypeRef` is better? `commonTypesRef` is very confusing.

Reply:
This I agree. commonTypesRef is a somewhat limiting and could be confused with ARM common types.